### PR TITLE
feat(api): Add iteration to issue and group filters

### PR DESCRIPTION
**Pull request by cristianocasella**
## Changes

This MR allow to filter issues from groups and projects by iteration.

### Documentation and testing

Please consider whether this PR needs documentation and tests. **This is not required**, but highly appreciated:

- [x] Documentation in the matching [docs section](https://github.com/python-gitlab/python-gitlab/tree/main/docs)
- [x] [Unit tests](https://github.com/python-gitlab/python-gitlab/tree/main/tests/unit) and/or [functional tests](https://github.com/python-gitlab/python-gitlab/tree/main/tests/functional)

This change doesn't include new tests and documentation as they don't exist for the other filters.  

Side note: The iteration_id filter is actually supported via API call but not documented yet on public gitlab documentation: https://docs.gitlab.com/ee/api/issues.html

### CODE REVIEW

File: gitlab/v4/objects/issues.py [hunk 1]
```diff
@@ -55,8 +55,10 @@ class IssueManager(RetrieveMixin, RESTManager):
         "state",
         "labels",
         "milestone",
+        "iteration_id",
         "scope",
         "author_id",
+        "iteration_id",
```
Comment (JohnVillalovos) [inline gitlab/v4/objects/issues.py:61]: Looks like this has been done twice. L58 above already does this.

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]
Thanks for the PR. One minor issue.

File: gitlab/v4/objects/issues.py (hunk 1 above)
Comment (cristianocasella) [reply gitlab/v4/objects/issues.py:61]: whops, thanks for finding it!

Comment (cristianocasella) [reply gitlab/v4/objects/issues.py:60]: I just removed the duplication!

**Review by JohnVillalovos — APPROVED** [review]

**Comment by JohnVillalovos** [comment]
Thanks for the PR!

**Comment by cristianocasella** [comment]
Thanks for the support and review @JohnVillalovos!

**Pull request merged**